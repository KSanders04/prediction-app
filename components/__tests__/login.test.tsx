import { render, fireEvent } from "@testing-library/react-native";
import { router } from "expo-router";
import Login from "../../app/login";

// Mock Firebase functions
jest.mock("@/components/firebaseFunctions", () => ({
  emailSignIn: jest.fn(),
}));

// Mock expo-router
jest.mock("expo-router", () => ({
  router: {
    push: jest.fn(),
    replace: jest.fn(),
  },
}));

describe("Login Component", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly", () => {
    const { getByText, getByPlaceholderText, getByTestId } = render(<Login />);

    expect(getByTestId("sign-in-text")).toBeTruthy();
    expect(getByTestId("sign-in-button")).toBeTruthy();
    expect(getByPlaceholderText("Email")).toBeTruthy();
    expect(getByPlaceholderText("Password")).toBeTruthy();
  });

  it("handles user input correctly", () => {
    const { getByPlaceholderText } = render(<Login />);

    const emailInput = getByPlaceholderText("Email");
    const passwordInput = getByPlaceholderText("Password");

    fireEvent.changeText(emailInput, "admin@admin.com");
    fireEvent.changeText(passwordInput, "12345678");

    expect(emailInput.props.value).toBe("admin@admin.com");
    expect(passwordInput.props.value).toBe("12345678");
  });

  it("shows error message when login fails", () => {
    const { getByTestId } = render(<Login />);
    const signInButton = getByTestId("sign-in-button");

    fireEvent.press(signInButton);
    // Add error message check once implemented
  });

  // Single snapshot test with proper cleaning
  it("matches snapshot", () => {
    const { toJSON } = render(<Login />);
    const snapshot = toJSON();

    // Clean the snapshot by removing dynamic properties
    const cleanSnapshot = (obj: any): Record<string, any> => {
      if (!obj || typeof obj !== 'object') return obj;
      
      const cleaned: Record<string, any> = Array.isArray(obj) ? [] : {};
      
      Object.entries(obj).forEach(([key, value]) => {
        if (!['nativeID', 'testID', 'style', 'key'].includes(key)) {
          cleaned[key] = cleanSnapshot(value);
        }
      });
      
      return cleaned;
    };

    expect(cleanSnapshot(snapshot)).toMatchSnapshot();
  });
});
