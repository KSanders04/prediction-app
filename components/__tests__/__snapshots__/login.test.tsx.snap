// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component matches snapshot 1`] = `
{
  "children": [
    {
      "children": [
        {
          "children": [
            {
              "children": [
                {
                  "children": [
                    "Sign In",
                  ],
                  "props": {},
                  "type": "Text",
                },
                {
                  "children": null,
                  "props": {
                    "autoCapitalize": "none",
                    "keyboardType": "email-address",
                    "onChangeText": [Function],
                    "placeholder": "Email",
                    "placeholderTextColor": "#3C4858",
                    "returnKeyType": "next",
                    "value": "",
                  },
                  "type": "TextInput",
                },
                {
                  "children": null,
                  "props": {
                    "autoCapitalize": "none",
                    "onChangeText": [Function],
                    "placeholder": "Password",
                    "placeholderTextColor": "#3C4858",
                    "returnKeyType": "done",
                    "secureTextEntry": true,
                    "value": "",
                  },
                  "type": "TextInput",
                },
                {
                  "children": [
                    {
                      "children": [
                        "Sign In",
                      ],
                      "props": {},
                      "type": "Text",
                    },
                  ],
                  "props": {
                    "accessibilityState": {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    },
                    "accessibilityValue": {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    },
                    "accessible": true,
                    "collapsable": false,
                    "focusable": true,
                    "onClick": [Function],
                    "onResponderGrant": [Function],
                    "onResponderMove": [Function],
                    "onResponderRelease": [Function],
                    "onResponderTerminate": [Function],
                    "onResponderTerminationRequest": [Function],
                    "onStartShouldSetResponder": [Function],
                  },
                  "type": "View",
                },
                {
                  "children": [
                    {
                      "children": [
                        "Don't have an account?",
                      ],
                      "props": {},
                      "type": "Text",
                    },
                    {
                      "children": [
                        {
                          "children": [
                            " Create Account",
                          ],
                          "props": {},
                          "type": "Text",
                        },
                      ],
                      "props": {
                        "accessibilityState": {
                          "busy": undefined,
                          "checked": undefined,
                          "disabled": undefined,
                          "expanded": undefined,
                          "selected": undefined,
                        },
                        "accessibilityValue": {
                          "max": undefined,
                          "min": undefined,
                          "now": undefined,
                          "text": undefined,
                        },
                        "accessible": true,
                        "collapsable": false,
                        "focusable": true,
                        "onClick": [Function],
                        "onResponderGrant": [Function],
                        "onResponderMove": [Function],
                        "onResponderRelease": [Function],
                        "onResponderTerminate": [Function],
                        "onResponderTerminationRequest": [Function],
                        "onStartShouldSetResponder": [Function],
                      },
                      "type": "View",
                    },
                  ],
                  "props": {},
                  "type": "View",
                },
              ],
              "props": {},
              "type": "View",
            },
          ],
          "props": {
            "accessibilityElementsHidden": undefined,
            "accessibilityLiveRegion": undefined,
            "accessibilityState": {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            },
            "accessible": false,
            "contentContainerStyle": {
              "alignItems": "center",
              "flexGrow": 1,
              "justifyContent": "center",
              "paddingBottom": 40,
            },
            "focusable": true,
            "importantForAccessibility": undefined,
            "keyboardShouldPersistTaps": "handled",
            "onClick": [Function],
            "onResponderGrant": [Function],
            "onResponderMove": [Function],
            "onResponderRelease": [Function],
            "onResponderTerminate": [Function],
            "onResponderTerminationRequest": [Function],
            "onStartShouldSetResponder": [Function],
            "showsVerticalScrollIndicator": false,
          },
          "type": "RCTScrollView",
        },
      ],
      "props": {
        "onLayout": [Function],
      },
      "type": "View",
    },
  ],
  "props": {},
  "type": "RCTSafeAreaView",
}
`;
